[TT-16761] docs: Updated docs for compression of APIs and Policies (#1810)
TykTechnologies/tyk-docs: planning-for-production/database-settings.mdx

In [MDCB](/api-management/mdcb) deployments, Data Plane Gateways store API definition backups in Redis to ensure resilience when the Control Plane is unavailable. For deployments with many or large API definitions, this can consume significant Redis memory.

> You can enable Zstandard compression to reduce the storage footprint by setting [`storage.compress_api_definitions`](/tyk-oss-gateway/configuration#storage-compress_api_definitions) and [`storage.compress_policies`](/tyk-oss-gateway/configuration#storage-compress_api_policies) to `true` in your Data Plane Gateway configuration:

Reviewer: ```suggestion
You can reduce the size of these backups using optional Zstandard compression. Typically you can expect to see approximately 75% compression, though this can vary depending on the specific configuration of your API definitions and policies.

Set [`storage.compress_api_definitions`](/tyk-oss-gateway/configuration#storage-compress_api_definitions) and [`storage.compress_policies`](/tyk-oss-gateway/configuration#storage-compress_api_policies) in your Data Plane Gateway configuration to compress your API definition and policy backups respectively:
```

```json
"storage": {

>   ...

Reviewer: ```suggestion
```

Don't need to indicate that this is a partial config.

  "compress_api_definitions": true,
  "compress_policies": true
}
```

> The Gateway will seamlessly read uncompressed API definitions and policies if they are already stored in Redis in that format. When the Data Plane is reloaded, the definitions and policies will be compressed before being stored.

Reviewer: ```suggestion
API definition compression is available from Tyk 5.12.0 onwards.
Policy compression is available from Tyk 5.13.0 onwards.

The Gateway will seamlessly read uncompressed API definitions and policies if they are already stored in Redis in that format. When the Data Plane is reloaded, the definitions and policies will be compressed before being stored.
```

> To mitigate the risk of decompression bombs, the maximum combined uncompressed size for the API definitions is limited to the value you configure for the `storage.max_decompressed_size`. Typically you should expect to see approximately 75% compression for Tyk OAS APIs.

Reviewer: Let's move the benefit (75% compression) higher up.

```suggestion
To mitigate the risk of decompression bombs, the maximum combined uncompressed size for the API definitions is limited to the value you configure for the `storage.max_decompressed_size`.
```
